docs/api/v1.md

```diff
@@ -0,0 +1,383 @@
+# Spice API
+## Version `v1`
+
+
+All routes are prefixed with `/api/v1/`
+
+
+Example: `GET localhost:1234/api/v1/processes`
+
+
+## Host System Info
+
+
+### `GET /filesystem/:path*`
+Gets the file(s) within the given path
+
+
+Path parameters:
+ - `path` {string}: URL encoded path to a file or directory.  If empty, refers to the root directory.
+
+
+Responses:
+ - `200`: Successfully got contents of path.
+    - Returns {[File](#file)}
+ - `400`: Invalid path.
+    - Returns {[Error](#error)}
+ - `403`: Insufficient privileges to open file.
+    - Returns {[Error](#error)}
+ - `404`: File doesn't exist.
+    - Returns {[Error](#error)}
+
+
+### `GET /processes`
+Gets the list of processes running on the host machine.
+
+
+Responses:
+ - `200`: Successfully got list of processes.
+    - Returns {[Process](#process)[]}
+ - `403`: Insufficient privileges to complete operation.
+    - Returns {[Error](#error)}
+
+
+# Debug
+All `/debug` routes are stateful--their responses are based on the currently attached process/executable.
+If the debugger is not attached, these endpoints respond with `409 Conflict`. None of the `/debug` routes respond with `409` for any other reason.
+
+
+### `POST /debug/attach/pid/:pid`
+Attach to a running process. If a process was already attached, detaches from that process.
+Path parameters:
+ - `pid` {integer}: PID of a running process
+
+
+Responses:
+ - `200`: Successfully attached to process.
+    - Returns {[AttachInfo](#attachInfo)} 
+ - `400`: Invalid PID
+     - Returns {[Error](#error)}
+ - `404`: Process with PID not found
+    - Returns {[Error](#error)}
+
+
+### `POST /debug/attach/bin/:path*`
+Attach to a binary, so it can be executed. Returns path to source code.
+
+
+Path parameters:
+ - `path` {string}: Path to binary to run and attach to.
```

Comment:
Yeah, I think we should add those as new fields. What would be the best format for the backend?